### PR TITLE
DynObj rework and various tests

### PR DESCRIPTION
**Pull request by kMutagene**
This PR will

- [x] Rework the `DynObj` API (parameter order, some new functions, XML docs)
- [x] Add various tests
  - [x] All instance methods of `DynamicObj()`
  - [x] All static methods from `DynObj`

### CODE REVIEW

**Comment by kMutagene** [comment]
Ready for review @HLWeil

File: src/DynamicObj/DynObj.fs [hunk 1]
```diff
@@ -63,39 +65,128 @@ module DynObj =
...
+    /// <summary>
+    /// Sets the given dynamic property name with the result of a mapping function applied to the given property value on the given DynamicObj if the value is Some('TPropertyValue).
+    /// If the given propertyValue is None, does nothing to the input DynamicObj.
+    /// </summary>
+    /// <param name="propertyName">The name of the dynamic property to set</param>
+    /// <param name="propertyValue">The value of the dynamic property to set</param>
+    /// <param name="mapping">A function to apply to the property value before setting it on the DynamicObj</param>
+    /// <param name="dynObj">The DynamicObj to set the property on</param>
+    /// <remarks>This function mutates the input DynamicObj</remarks>
+    let setValueOptBy (propertyName: string) (propertyValue: 'TPropertyValue option) (mapping: 'TPropertyValue -> 'UPropertyValue) (dynObj: DynamicObj) =
```
Comment (HLWeil) [inline src/DynamicObj/DynObj.fs:137]: This seems oddly specific

File: tests/DynamicObject.Tests/DynObjStaticMethods.fs
Comment (HLWeil) [inline tests/DynamicObject.Tests/DynObjStaticMethods.fs:1]: Why not just call this file `DynObj`? If I see correctly, `StaticMethods` is wrong anyways as you're testing functions here.

**Review by HLWeil — CHANGES_REQUESTED** [review]
Great stuff!
But I don't agree with the tests rename. I think `DynamicObj` and `DynObj` would both better represent the functionality and the files of interest.

File: tests/DynamicObject.Tests/DynObjStaticMethods.fs
Comment (kMutagene) [reply tests/DynamicObject.Tests/DynObjStaticMethods.fs:1]: done

File: src/DynamicObj/DynObj.fs (hunk 1 above)
Comment (kMutagene) [reply src/DynamicObj/DynObj.fs:137]: and used hundreds of times in Plotly.NET

**Pull request merged**